src/components/workspace/editorMode/source.js

```diff
@@ -0,0 +1,820 @@
+ace.define("ace/mode/doc_comment_highlight_rules",["require","exports","module","ace/lib/oop","ace/mode/text_highlight_rules"], function(acequire, exports, module) {
```

Comment:
It is probably good to add some descriptions on how is this custom mode generated and how it is different from its original version.

Reply:
Yep. I have just added the description.

Reply:
It seems like these three name are used in all ace.define functions. I haven't figured out the exact meaning of these names yet.